src/irmin/conf.ml

```diff
@@ -141,6 +142,32 @@ let get (_, d) k =
 let keys (_, conf) = M.to_seq conf |> Seq.map (fun (k, _) -> k)
 let with_spec (_, conf) spec = (spec, conf)
 
+let pp ppf (_, conf) =
+  M.to_seq conf
+  |> Seq.map (fun (K k, v) ->
+         ( k.name,
+           match k.of_univ v with
+           | Some v -> Type.to_string k.ty v
+           | None -> assert false ))
+  |> List.of_seq
+  |> List.sort_uniq (fun (x, _) (y, _) -> String.compare x y)
+  |> Fmt.Dump.(list (pair string string)) ppf
+
+let equal t1 t2 =
+  t1 == t2
+  ||
+  let to_seq (_, conf) =
+    M.to_seq conf
+    |> Seq.map (fun (K k, v) ->
+           ( k.name,
+             match k.of_univ v with
+             | Some v -> Type.to_string k.ty v
+             | None -> assert false ))
+  in
```

Comment:
I think it'd be nice to extract this `to_seq` function to remove the duplicated code from `pp` and `equal`. WDYT?

Reply:
good idea, done!